.github/workflows/tf-apply-staging.yml

```diff
@@ -0,0 +1,48 @@
+name: Terraform Apply Staging
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "terragrunt/aws/**"
+      - "terragrunt/env/staging/**"
+      - "terragrunt/env/common/**"
+      - "terragrunt/env/root.hcl"
+      - ".github/workflows/tf-apply-staging.yml"
+
+env:
+  AWS_REGION: ca-central-1
+  TERRAFORM_VERSION: 1.11.4
+  TERRAGRUNT_VERSION: 0.77.22
+  
+permissions:
+  id-token: write
+  contents: read
+  pull-requests: write
+  actions: write
+  checks: write
+  statuses: write
+
+jobs:
+  terragrunt-apply:
+    if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: setup terraform tools
+        uses: cds-snc/terraform-tools-setup@v1
+
+      - name: configure aws credentials using OIDC
+        uses: aws-actions/configure-aws-credentials@ececac1a45f3b08a01d2dd070d28d111c5fe6722 # v4.1.0
+        with:
+          role-to-assume: arn:aws:iam::211125499457:role/cra-upd-dashboard-apply
```

Comment:
@sylviamclaughlin Not sure if this is why the workflow failed, but I just "pretended" that this role exists, using the same naming convention  I saw in other CDS repos. And also did the same for the `plan` workflow.